### ISSUE TITLE
ENHANCEMENT: Support custom surface exposure types (PHPP v10 custom exposure zones)

### ISSUE THREAD

**Issue by ed-p-may** [PH-Tools/honeybee_ph#78]
## Summary

Surfaces bounding an **attached, unconditioned zone** (an attached garage, an unheated sunspace, a buffer stair, an adjacent unheated tenant space) need a custom exposure rather than `Outdoors` / `Ground` / `Surface`. The WUFI-Passive half of this already works end-to-end via `PhAdditionalZone`. The **PHPP half does not exist** — PHPP v10's custom exposure zones have no export path, and it has never been confirmed whether `PhAdditionalZone` carries everything PHPP would need.

This issue tracks the `honeybee_ph` side of that gap. The export itself belongs to PH-Tools/PHX#65.

## Current behavior

**What already works.** `PhAdditionalZone` extends honeybee-energy's `OtherSideTemperature` with `zone_name`, `zone_type`, `zone_id_num` and `temperature_reduction_factor`:

https://github.com/PH-Tools/honeybee_ph/blob/1bcdc92741609d942190074eee47931e0951e062/honeybee_energy_ph/boundarycondition.py#L24-L59

It is set from Grasshopper (`honeybee_ph_rhino/gh_compo_io/addnl_zone.py` → the `HBPH - PH Additional Zone` component), and `PHX/from_HBJSON/create_building.py` maps it to WUFI's `SURFACE` exposure, colors the components as `INNER_WALL`, and builds a PHX zone per unique BC.

**What is missing.**

1. **No PHPP path.** Nothing writes custom exposure zones to a PHPP v10 workbook. Whether `temperature_reduction_factor` alone is sufficient for PHPP's model, or whether additional attributes are required, is unverified.
2. **The BC is registered by monkey-patching.** `PhAdditionalZone` is injected into the `honeybee.boundarycondition` namespace at import time, which is import-order sensitive — the reason for the `IMPORTANT` warning at the top of the file:

https://github.com/PH-Tools/honeybee_ph/blob/1bcdc92741609d942190074eee47931e0951e062/honeybee_energy_ph/_extend_honeybee_energy_ph.py#L193-L197

## Desired behavior

1. Confirm what PHPP v10's custom exposure zone actually requires as input, and extend `PhAdditionalZone` to carry any attribute it needs that is not there today (serialized backward-compatibly per the repo's rules).
2. If no new attributes are needed, say so explicitly in this issue and hand the work entirely to PH-Tools/PHX#65 — a "nothing required here" answer is a valid and useful outcome.
3. Longer term, rebuild `PhAdditionalZone` on a sanctioned honeybee boundary-condition extension mechanism if one lands upstream (see ladybug-tools/honeybee-energy#1049), and drop the monkey-patch.

## Affected code

- [`honeybee_energy_ph/boundarycondition.py`](https://github.com/PH-Tools/honeybee_ph/blob/1bcdc92741609d942190074eee47931e0951e062/honeybee_energy_ph/boundarycondition.py) — `PhAdditionalZone`
- [`honeybee_energy_ph/_extend_honeybee_energy_ph.py#L193-L197`](https://github.com/PH-Tools/honeybee_ph/blob/1bcdc92741609d942190074eee47931e0951e062/honeybee_energy_ph/_extend_honeybee_energy_ph.py#L193-L197) — BC registration monkey-patch
- Downstream: `PHX/from_HBJSON/create_building.py` (WUFI mapping, already working) and the unwritten PHPP writer
- Downstream: `honeybee_ph_rhino/gh_compo_io/addnl_zone.py` (`HBPH - PH Additional Zone` component) — only if new attributes are added

## Acceptance criteria

- [ ] The PHPP v10 custom-exposure input requirements are documented in this issue (which cells / which worksheet, and which values they need).
- [ ] Any attribute PHPP requires that `PhAdditionalZone` does not already carry is added, with a default in `__init__`, written in `to_dict()`, read via `.get(key, default)` in `from_dict()`, and copied on duplicate — old HBJSON still loads.
- [ ] A `PhAdditionalZone` BC survives a full HBJSON write → reload cycle with all attributes intact (regression test).
- [ ] If new attributes were added, the `HBPH - PH Additional Zone` Grasshopper component exposes them.
- [ ] `python3 -m pytest` passes at 100% coverage.

## Related

- **PH-Tools/PHX#65** — "ENHANCEMENT: Custom Exposures". Owns the actual WUFI + PHPP export work; this issue is its `honeybee_ph` data-model counterpart.
- **ladybug-tools/honeybee-energy#1049** — "Extending HB-E Boundary Conditions?". Shared root cause: honeybee has no sanctioned extension point for adding a boundary condition that serializes and round-trips cleanly, which is why `PhAdditionalZone` relies on a monkey-patch. If a proper mechanism lands upstream, `PhAdditionalZone` should be rebuilt on it.

**Comment by ed-p-may** [comment]
## PHPP v10 custom-exposure input requirements (acceptance criterion 1)

Answered against the PHPP 10.6 workbook structure and the PHPP 10 manual chapter for `Areas` ([[P16]] §16.2.1 to §16.2.3). Verified cell-by-cell; addresses below are from the blank 10.6 workbook.

**The short answer: "nothing required here" is not the outcome.** `PhAdditionalZone` carries one `temperature_reduction_factor`. PHPP wants five per zone, and it caps the number of zones at three.

### Where the input lives

Two separate selectors meet at the `Areas` summary block.

**1. The area group.** `Areas` summary rows 9 to 27 define the area groups. Rows 19, 20 and 21 are groups 12, 13 and 14, the three user-definable groups. Each carries a user-typed description in `M19:M21` and a temperature-zone letter in `K19:K21`.

**2. The temperature zone.** Column `K` of the summary is the zone code. PHPP 10 defines seven, in the lookup table `Areas!BL9:BM15`:

| Code | Factor source | Meaning |
|---|---|---|
| `A` | `1` | Ambient air |
| `B` | `Ground!P116` | Ground / unheated basement |
| `P` | `Ground!P116` | Perimeter |
| `I` | `0` | Neighbouring building, no heat loss |
| `X` | `IF(Areas!Z19="",1,Areas!Z19)` | **User-defined zone 1** |
| `Y` | `IF(Areas!Z20="",1,Areas!Z20)` | **User-defined zone 2** |
| `Z` | `IF(Areas!Z21="",1,Areas!Z21)` | **User-defined zone 3** |

An area row resolves its factor through a hidden chain: the group selector `Areas!M41` (and down), `BJ41 = VLOOKUP`-able group number, then `BK41 = VLOOKUP(BJ41, $BJ$9:$BK$27, 2, FALSE)`, which lands on the summary row's zone letter and through it on the table above.

### The five factors

This is the part the current model cannot express. The input block is `Areas!Y16:AF21`, headed "User-defined temperature reduction factors". Zone labels `X`/`Y`/`Z` are fixed in `Y19:Y21`; the five factor columns are all user input:

| Column | Factor |
|---|---|
| `Z19:Z21` | Heating demand |
| `AA19:AA21` | Heating load |
| `AC19:AC21` | Cooling demand |
| `AE19:AE21` | Cooling load |
| `AF19:AF21` | Passive cooling |

They are consumed by different worksheets, so they are genuinely five values and not one value shown five times. The heating-demand column is the one the `Areas` per-surface chain reads. The cooling-demand column is read straight off by `Verification!AR64:AS69`.

Blank means 1.0, which matches the current `temperature_reduction_factor` default.

### The auxiliary calculation, and why it stays out of this repo

`Areas!Y22:AF24` derives all five factors from a single input, `AF23`, the adjacent zone's temperature in °C. Per [[P16]] §16.2.3, adjacent-zone temperature alone is sufficient, and separate factors need entering only when the summer and winter adjacent-zone temperatures differ.

That auxiliary calculation cannot move into honeybee-ph. Its formulas read `Verification!K28`, `Climate!R42`, `Climate!R60`, `Heating load!D10:D11` and `Cooling load!L10:M10`, all of which are PHPP-computed. Derivation belongs to the PHX writer (PH-Tools/PHX#65) or to the modeller.

PHI's definition of the factor, paraphrased from §16.2.3: the ratio of heat flow toward the adjacent zone to the heat flow that would occur toward outdoor air. Values above 1 or below 0 can be legitimate, so the field must not be range-clamped to 0 to 1.

### Gap against `PhAdditionalZone`

| PHPP needs | Model today |
|---|---|
| Zone name (`Areas!M19:M21`) | `zone_name` ✅ |
| Heating-demand factor (`Z19:Z21`) | `temperature_reduction_factor` ✅ |
| Heating-load factor (`AA19:AA21`) | missing |
| Cooling-demand factor (`AC19:AC21`) | missing |
| Cooling-load factor (`AE19:AE21`) | missing |
| Passive-cooling factor (`AF19:AF21`) | missing |
| Adjacent-zone temperature (`AF23`, optional) | missing as a PH field |
| At most three zones per model | unbounded |

Note the last row. Nothing in honeybee-ph limits how many distinct `PhAdditionalZone` boundary conditions a model carries, and PHPP has exactly three slots. That constraint cannot be enforced here without breaking the WUFI path, which has no such limit, so it belongs in the PHX PHPP writer as a validation error. Worth stating explicitly in PH-Tools/PHX#65 rather than discovering it on a real model.

### Proposed change here

Four new optional factors plus the adjacent-zone temperature, all defaulting to `None` per the repo's absence convention (`context/UBIQUITOUS_LANGUAGE.md` line 169, absence is `None`, never `0`):

```
temperature_reduction_factor          float  = 1.0    # existing; heating demand
heating_load_reduction_factor         float? = None
cooling_demand_reduction_factor       float? = None
cooling_load_reduction_factor         float? = None
passive_cooling_reduction_factor      float? = None
adjacent_zone_temperature_c           float? = None
```

`None` on a factor means "use the heating-demand factor", which reproduces today's behavior for every existing model and matches the common case where the adjacent zone has one temperature year-round. Keeping `temperature_reduction_factor` as the heating-demand slot avoids a rename and keeps old HBJSON loading unchanged.

`adjacent_zone_temperature_c` is carried but not used here. It is the input a modeller actually has, and PHX needs it to reproduce the auxiliary calculation.

Serialization follows the standard four steps: default in `__init__`, written in `to_dict()`, read with `.get(key, default)` in `from_dict()`, copied in `duplicate()`. All five are new fields, so `.get()` is required.

### Remaining acceptance criteria

- [x] PHPP v10 custom-exposure input requirements documented (this comment)
- [ ] Add the five fields to `PhAdditionalZone` with backward-compatible serialization
- [ ] Round-trip regression test for a `PhAdditionalZone` BC through a full HBJSON write and reload
- [ ] Expose the new inputs on the `HBPH - PH Additional Zone` Grasshopper component
- [ ] `python3 -m pytest` passes, coverage at or above the 75% floor

The monkey-patch item stays out of scope. It is blocked on ladybug-tools/honeybee-energy#1049 and is unrelated to the PHPP gap.

### Hand-off to PH-Tools/PHX#65

- Write `Areas!Z19:AF21` from at most three distinct `PhAdditionalZone` BCs; raise a clear error on the fourth.
- Write the zone letter into the summary row `K19:K21` and the zone name into `M19:M21`.
- Assign each surface to group 12, 13 or 14 through the `Areas!M` group selector.
- Where only `adjacent_zone_temperature_c` is given, either write it to `AF23` and let PHPP's auxiliary calculation run, or reproduce that calculation. Writing the temperature is the simpler and more faithful option.

Sources: PHPP 10.6 workbook structure (`Areas` sheet, cell addresses only, no licensed values reproduced) and PHPP 10 manual [[P16]] §16.2.1 to §16.2.3.

**Comment by ed-p-may** [comment]
## Follow-up: consumers traced, and a second constraint found

The `phi-rules` PHPP 10 corpus has been audited and corrected behind this issue, which firmed up two things in the comment above and added one.

### Consumers, now traced rather than inferred

I searched every formula in the 10.6 workbook for references to each of the five factor columns. They have **disjoint** consumer sets, which settles that they are five real values:

| Column | Factor | Consumed by |
|---|---|---|
| `Z19:Z21` | heating demand | `Areas!BM12:BM14` (the per-surface chain), `Annual heating!K12:K23`, `Heating`, `Verification` |
| `AA19:AA21` | heating load | `Heating load!J29:J41` only |
| `AC19:AC21` | cooling demand | `Cooling`, `Verification!AR64:AS69` and `BE63` |
| `AE19:AE21` | cooling load | `Cooling load!J19:J31` only |
| `AF19:AF21` | passive cooling | `Summer!K13:K24` only |

Each dropped factor therefore shows up as a wrong number on a different worksheet, which is the awkward failure mode: no single check catches all five.

### `Heating load` reads two different factors for the same zone

Worth knowing before anyone writes the PHX side. `Heating load` has two parallel transmission blocks, and they do not agree by design:

- the visible block's `J29:J41` takes the **heating-load** factor from `Areas!AA19:AA21`
- the parallel room-loss block at `AC29:AC41` takes the **heating-demand** factor, via `'Annual heating'!K12:K23` which resolves to `Areas!Z19:Z21`

PHPP labels that second column `AC28` literally `Always 1 (except for X,Y,Z)`. So a custom zone whose heating-demand and heating-load factors differ produces a room transmission loss that does not scale from the whole-building load. That is intended behavior, not a workbook defect, and a port that collapses the two will be wrong in one of the two places.

### Nothing changes in the proposal

The five fields plus `adjacent_zone_temperature_c` still cover it, and `None` still means "use the heating-demand factor". The tracing only raises confidence that carrying one factor is genuinely lossy rather than theoretically lossy.

Corpus commits, for anyone checking the derivation: `fc8814e`, `7e84151`, `2f6af65`, `3182a0e` in `bldgtyp/phi-rules`. The `Areas` cell map now carries the full temperature-zone mechanism, which it did not when this issue was filed.
